### ISSUE TITLE
API is missing implementations of AuthCredential

### ISSUE THREAD

**Issue by marcojakob** [firebase/firebase-js-sdk#120]
### Environment

  * Operating System version: any
  * Firebase SDK version: 4.1.3
  * Firebase Product: auth

### Problem

The API reference is missing details about implementations of the [`AuthCredential`](https://firebase.google.com/docs/reference/js/firebase.auth.AuthCredential) interface. There is an implementation specific to each auth provider. The attribute `accessToken`, and for Twitter also the attribute `secret` of the `AuthCredential` needs to be accessed but it is not documented in the API. The only place where it is mentioned is in the example code, for example in [GithubAuthProvider](https://firebase.google.com/docs/reference/js/firebase.auth.GithubAuthProvider):

To get a hold of the `accessToken` of Facebook, Github and Google and to get the `accessweespecially for the following case we need to know the details about the implementation:

```JavaScript
    // This gives you a GitHub Access Token.
    var token = result.credential.accessToken;
```

The reason I'm opening this issue is because we need to rely on the official JS API for the implementation on the [Dart side of Firebase](https://github.com/firebase/firebase-dart) to be able to resolve [this issue](https://github.com/firebase/firebase-dart/issues/118).

**Comment by jshcrowthe** [comment]
Seems like a good place to improve our documentation. @bojeil-google is this something you have cycles for?

**Comment by bojeil-google** [comment]
Yeah, I think there a few tweaks needed to the auth externs. I will try to get it done this week.

**Comment by MeghaB** [comment]
For internal tracking, the documentation updates in b/156548881
